### PR TITLE
Update nomad operator snapshot inspect with more detail

### PR DESCRIPTION
**Pull request by nvanthao**
Continue from https://github.com/hashicorp/nomad/pull/18372

Sample output

```
Created  2024-03-02T08:50:27+11:00
ID       2-7-1709329827483
Size     1021 B
Index    7
Term     2
Version  1

Type             Count  Size
----             -----  ----
SchedulerConfig  1      242 B
NodePool         2      236 B
Namespace        1      209 B
RootKeyMeta      1      127 B
Index            3      75 B
ClusterMetadata  1      71 B
TimeTable        1      61 B
----             -----  ----
Total            -      1021 B
```

### CODE REVIEW

File: nomad/fsm.go [hunk 1]
```diff
@@ -840,9 +872,6 @@ func (n *nomadFSM) handleJobDeregister(index uint64, jobID, namespace string, pu
 
 		stopped := current.Copy()
 		stopped.Stop = true
-		if submitTime != 0 {
-			stopped.SubmitTime = submitTime
-		}
```
[the remark this answers is not in the document]
Comment (nvanthao) [reply nomad/fsm.go]: apologies @schmichael I have corrected the conflict. Please let me know how it goes.

File: nomad/fsm.go [hunk 2]
```diff
@@ -3212,3 +3245,47 @@ func (s *nomadSnapshot) persistJobSubmissions(sink raft.SnapshotSink, encoder *c
...
+func (s SnapshotType) String() string {
+	v, ok := snapshotTypeStrings[s]
+	if ok {
+		return v
+	}
+	v, ok = enterpriseSnapshotType(s)
+	if ok {
+		return v
+	}
+	return "others"
```
Comment (schmichael) [inline nomad/fsm.go:3290]: Sorry I didn't notice this before. Let's match Consul here. I think that's this: https://github.com/hashicorp/consul/blob/v1.18.0/agent/structs/structs.go#L3065
```suggestion
	return fmt.Sprintf("Unknown(%d)", s)
```

**Comment by schmichael** [comment]
For Hashicorporeals ( :wave:  @jrasell) looking this, there is a corresponding PR at hashicorp/nomad-enterprise#1398 for the enterprise types strings.

**Review by schmichael — APPROVED** [review]

**Review by jrasell — APPROVED** [review]
Hi @nvanthao and thanks for this work.

I've made a couple of minor changes to get it ready for merging, which I will time with our enterprise PR.

Thanks again!

**Pull request merged**